dev/com.ibm.ws.ejbcontainer.timer.persistent_fat/test-applications/MissedTimerActionWeb.war/src/com/ibm/ws/ejbcontainer/timer/persistent/missed/web/MissedTimerActionServlet.java

```diff
@@ -0,0 +1,188 @@
+    }
+
+    /**
+     * Test Persistent Timer missed action default behavior when failover has been enabled.
+     * The default behavior with failover should be ONCE. <p>
+     *
+     * This test will confirm the following :
+     * <ol>
+     * <li> Interval timer will skip expirations missed because of a delay.
+     * <li> Timer.getNextTimeout() will return values in the future; skipping missed expirations.
+     * </ol>
+     */
+    public void testMissedTimerActionDefaultWithFailover() throws Exception {
+        // Default when failover is enabled is ONCE
+        testMissedTimerAction("testMissedTimerActionDefaultWithFailover", "ONCE");
+    }
+
+    /**
+     * Test Persistent Timer missed action "ALL" behavior when failover has not been enabled. <p>
+     *
+     * This test will confirm the following :
+     * <ol>
+     * <li> Interval timer will run all expirations despite a delay that causes several to be missed.
+     * <li> Timer.getNextTimeout() will return values in the past for missed expirations.
+     * </ol>
+     */
+    public void testMissedTimerActionAllNoFailover() throws Exception {
+        testMissedTimerAction("testMissedTimerActionAllNoFailover", "ALL");
+    }
+
+    /**
+     * Test Persistent Timer missed action "ALL" behavior when failover has been enabled. <p>
+     *
+     * This test will confirm the following :
+     * <ol>
+     * <li> Interval timer will run all expirations despite a delay that causes several to be missed.
+     * <li> Timer.getNextTimeout() will return values in the past for missed expirations.
+     * </ol>
+     */
+    public void testMissedTimerActionAllWithFailover() throws Exception {
+        testMissedTimerAction("testMissedTimerActionAllWithFailover", "ALL");
+    }
+
+    /**
+     * Test Persistent Timer missed action "ONCE" behavior when failover has not been enabled. <p>
+     *
+     * This test will confirm the following :
+     * <ol>
+     * <li> Interval timer will skip expirations missed because of a delay.
+     * <li> Timer.getNextTimeout() will return values in the future; skipping missed expirations.
+     * </ol>
+     */
+    public void testMissedTimerActionOnceNoFailover() throws Exception {
+        testMissedTimerAction("testMissedTimerActionOnceNoFailover", "ONCE");
+    }
+
+    /**
+     * Test Persistent Timer missed action "ONCE" behavior when failover has been enabled. <p>
+     *
+     * This test will confirm the following :
+     * <ol>
+     * <li> Interval timer will skip expirations missed because of a delay.
+     * <li> Timer.getNextTimeout() will return values in the future; skipping missed expirations.
+     * </ol>
+     */
+    public void testMissedTimerActionOnceWithFailover() throws Exception {
+        testMissedTimerAction("testMissedTimerActionOnceWithFailover", "ONCE");
+    }
+
+    /**
+     * Test Persistent Timer missed action "ONCE" behavior when failover has not been enabled. <p>
```

Comment:
And again, "NONE"

Reply:
Will correct with next test update that will add server restart variations.